dask/utils.py

```diff
@@ -1125,11 +1125,18 @@ def is_arraylike(x):
     """
     from .base import is_dask_collection
 
+    is_duck_array = hasattr(x, "__array_function__") or hasattr(x, "__array_ufunc__")
```

Comment:
@pentschev do you have thoughts on this formulation?

Reply:
xref-ing @pentschev's earlier comment https://github.com/dask/dask/pull/7321#discussion_r587902321

Reply:
The duck arrays I'm aware of (i.e. NumPy, sparse, cupy, pint, unyt) all will result in `is_duck_array` being `True`. That said, I can't be 100% certain that this covers all array-like libraries that are used with Dask today. Perhaps this is a good opportunity for us to specify the conditions for which an array library will work with Dask Array. Xarray has a nice documentation section on this http://xarray.pydata.org/en/stable/internals.html#internals-duck-arrays topic